app/queries/student_details_resolver.rb

```diff
@@ -50,7 +50,7 @@ def targets_pending_review
   end
 
   def current_course_targets
-    course.targets.live.joins(:level).where.not(levels: { number: 0 })
+    @current_course_targets ||= course.targets.live.joins(:level).where.not(levels: { number: 0 })
```

Comment:
This memoization doesn't help since it'd save an `ActiveRecord::Relation`, not data loaded from DB.